### PR TITLE
[UNR-5651] 4.27 compatibility

### PR DESCRIPTION
**Pull request by mironec**

### CODE REVIEW

File: Game/Source/GDKShooter/Private/Characters/GDKCharacter.cpp [hunk 1]
```diff
@@ -176,7 +176,11 @@ FGenericTeamId AGDKCharacter::GetGenericTeamId() const
 	return TeamComponent->GetTeam();
 }
 
+#if ENGINE_MINOR_VERSION < 27
```
Comment (simonsarginson) [inline Game/Source/GDKShooter/Private/Characters/GDKCharacter.cpp:179]: dont we want to check the full version (major + minor) here?

Comment (mironec) [reply Game/Source/GDKShooter/Private/Characters/GDKCharacter.cpp:179]: Hm, true, with UE5 coming, I guess.
So I guess when doing `< MINOR`, we should do `<MINOR && MAJOR==4` and when doing `>=MINOR` we should do `>=MINOR||MAJOR==5`? Or stick to one style of checking?

Comment (mironec) [reply Game/Source/GDKShooter/Private/Characters/GDKCharacter.cpp:179]: wanted to formalize it for a while anyway

Comment (simonsarginson) [reply Game/Source/GDKShooter/Private/Characters/GDKCharacter.cpp:179]: Rider does this for their plugin which I think is pretty straightforward:
`#if ENGINE_MAJOR_VERSION == 4 && ENGINE_MINOR_VERSION <= 24`

Comment (simonsarginson) [reply Game/Source/GDKShooter/Private/Characters/GDKCharacter.cpp:179]: Oh yeah the `>=` case is not nice you could just use the else in those cases?
```
#if ENGINE_MAJOR_VERSION == 4 && ENGINE_MINOR_VERSION <= 24
// nothing
#else
something()
#endif
```

**Review by simonsarginson — APPROVED** [review]

File: .buildkite/nightly.steps.yaml [hunk 2]
```diff
@@ -8,12 +8,12 @@ steps:
...
     build:
       branch: *ci_version
-      message: "exampleproject-4.26 ${BUILDKITE_MESSAGE}"
+      message: "exampleproject-4.27 ${BUILDKITE_MESSAGE}"
       env: 
         BUILD_TYPE: "EXAMPLE"
-        GDK_BRANCH: "${GDK_BRANCH:-match_branch_name_pref::master}"
-        ENGINE_BRANCH: "${ENGINE_BRANCH_426:-match_branch_name_pref_engine::4.26-SpatialOSUnrealGDK}"
-        ENGINE_MAJOR: "4.26"
+        GDK_BRANCH: "${GDK_BRANCH:-match_branch_name_pref::feature/4.27-support}"
```
Comment (joshuahuburn) [inline .buildkite/nightly.steps.yaml:14]: Change all values to proper ones

File: Game/Source/GDKShooter/Public/Weapons/Weapon.h [hunk 3]
```diff
@@ -28,73 +34,73 @@ class GDKSHOOTER_API AWeapon : public AHoldable
...
 	UPROPERTY(EditAnywhere, BlueprintReadOnly)
-	FVector2D ReticleSize = FVector2D(16, 16);
+		FVector2D ReticleSize = FVector2D(16, 16);
 
 	virtual void ForceCooldown(float Cooldown) override;
 
 	//Recoil applied as input to the CharacterController rotations
 	UPROPERTY(EditAnywhere, BlueprintReadOnly)
-	float RecoilYawMax;
+		float RecoilYawMax;
```
Comment (joshuahuburn) [inline Game/Source/GDKShooter/Public/Weapons/Weapon.h:51]: Undo these whitespace changes

**Pull request merged**